php/libraries/Utility.class.inc

```diff
@@ -154,6 +155,39 @@ class Utility
         return $list;
     }
 
+    /**
+     * Returns a list of sites by site IDs from the database
+     *
+     * @param array $siteIDs a list of the study center IDs
+     *
+     * @return array an associative array("center ID" => "site name")
+     */
+    static function getSiteNameListByIDs(array $siteIDs): array
+    {
+        $factory = NDB_Factory::singleton();
+        $DB      = $factory->database();
+
+        $params = [];
+        $values = [];
+        foreach ($siteIDs as $key => $siteID) {
+            $para          = "cid$key";
+            $params[]      = ":$para";
+            $values[$para] = $siteID;
+        }
+
+        $sites = $DB->pselect(
+            "SELECT CenterID as ID, Name FROM psc WHERE CenterID in (" .
+            implode(', ', $params) . ")",
+            $values
+        );
+
+        $list = [];
+        foreach ($sites as $site) {
+            $list[$site['ID']] = $site['Name'];
+        }
```

Comment:
```suggestion
        $list = $DB->pselectColWithIndexKey(
            "SELECT CenterID as ID, Name FROM psc WHERE CenterID in (" .
            implode(', ', $params) . ")",
            $values,
            "ID"
        );
```

Reply:
OK, if you are sure, I am not able to test now. I will submit a commit

Reply:
not manually tested though, because of a problem I just reported: 6850